internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -546,9 +547,18 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		return false, err
 	}
 
+	var mirrorRegistryConfiguration *common.MirrorRegistryConfiguration
+	if infraEnvInternal.MirrorRegistryConfiguration != "" {
```

Comment:
If we're only doing this for the internal ignition override, can we move this into the conditional?

Also generally I'm not thrilled with this approach, but I don't have a better alternative for now without making some significant unrelated changes to how we save the internal ignition so I suppose we'll just leave it for now.